lib/Driver/Compilation.cpp

```diff
@@ -458,6 +458,26 @@ namespace driver {
       }
     }
 
+    /// Check to see if a batch-constituent job produced a zero-length
+    /// serialized diagnostics file, which is used to indicate constituents that
+    /// were batched together with a failing constituent but did not,
+    /// themselves, produce any errors.
+    bool batchConstituentJobProducedZeroLengthSerializedDiagnostics(
+      const Job *J) const {
+      auto DiaPath =
+        J->getOutput().getAnyOutputForType(file_types::TY_SerializedDiagnostics);
+      if (!DiaPath.empty()) {
+        if (llvm::sys::fs::is_regular_file(DiaPath)) {
```

Comment:
Same comment as above-- I'd be inclined to collapse and reverse this and prior into an or:
"if (<path-is-empty> || <is-not-regular-file>) return false;"

Reply:
fixed

Reply:
Great!